src/AllUserSleep.js

```diff
@@ -1,7 +1,87 @@
+var dayjs = require("dayjs");
+var duration = require('dayjs/plugin/duration')
+dayjs.extend(duration);
+var isBetween = require('dayjs/plugin/isBetween')
+dayjs.extend(isBetween)
+
 class AllUserSleep {
   constructor(sleepData) {
     this.sleepData = sleepData;
   }
+
+  calcAvgSleepQuality(sleepData) {
```

Comment:
Nice `reduce` here as well!

Similar to my comment below, this looks similar to the `calcAvgSleep` reduce in `UserSleep`. I wonder if this could turn into a `calculateAverage` helper method that takes in `data` and `property` as parameters, and then returns an average of any given property in a specific data set (whether this is for a single user or all users!)